app/models/concerns/products/scheduling_support.rb

```diff
@@ -77,7 +77,11 @@ def quick_reservation_intervals
   end
 
   def offline?
-    current_offline_reservations.present?
+    if current_offline_reservations.loaded?
+      current_offline_reservations.present?
+    else
+      current_offline_reservations.exists?
+    end
```

Comment:
Just out of curiosity, why can't we use `exists?` in all cases?

Reply:
if the association is loaded then `exists?` will make a db query whereas `present?` won't